multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/PollStartAppExecutionWithRollback.java

```diff
@@ -0,0 +1,56 @@
+package org.cloudfoundry.multiapps.controller.process.steps;
+
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudApplicationExtended;
+import org.cloudfoundry.multiapps.controller.core.cf.CloudControllerClientFactory;
+import org.cloudfoundry.multiapps.controller.core.model.IncrementalAppInstanceUpdateConfiguration;
+import org.cloudfoundry.multiapps.controller.core.security.token.TokenService;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+
+import com.sap.cloudfoundry.client.facade.CloudControllerClient;
+import com.sap.cloudfoundry.client.facade.domain.CloudApplication;
+
+public class PollStartAppExecutionWithRollback extends PollStartAppStatusExecution {
+
+    public PollStartAppExecutionWithRollback(CloudControllerClientFactory clientFactory, TokenService tokenService) {
+        super(clientFactory, tokenService);
+    }
+
+    @Override
+    public AsyncExecutionState execute(ProcessContext context) {
+        AsyncExecutionState asyncExecutionState = super.execute(context);
+        if (asyncExecutionState == AsyncExecutionState.ERROR) {
+            rollbackOldAppInstances(context);
+        }
+        return asyncExecutionState;
```

Comment:
if the step is Done it will be increase by 1 in the parent step
if the step is in Error state it will trigger old application instance update and if the process is retried it will wait for the old app to scale up